pkg/cedar/core/software_products.go

```diff
@@ -0,0 +1,84 @@
+package cedarcore
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/guregu/null/zero"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/models"
+
+	software_products "github.com/cmsgov/easi-app/pkg/cedar/core/gen/client/software_products"
+)
+
+// GetBudgetBySystem queries CEDAR for budget information (NJD EXPAND ON THIS?) associated with a particular system, taking the version-independent ID of a system
+func (c *Client) GetSoftwareProductsBySystem(ctx context.Context, cedarSystemID string) (*models.CedarSoftwareProducts, error) {
+	if !c.cedarCoreEnabled(ctx) {
+		appcontext.ZLogger(ctx).Info("CEDAR Core is disabled")
+		return nil, nil
+	}
+	cedarSystem, err := c.GetSystem(ctx, cedarSystemID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Construct the parameters
+	params := software_products.NewSoftwareProductsFindListParams()
+	params.SetID(cedarSystem.VersionID)
+	params.HTTPClient = c.hc
+
+	// Make the API call
+	resp, err := c.sdk.SoftwareProducts.SoftwareProductsFindList(params, c.auth)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.Payload == nil {
+		return nil, fmt.Errorf("no body received") // NJD - this ok here?
+	}
+
+	// Convert the auto-generated struct to our own pkg/models struct
+
+	// Start by converting child SoftwareProduct object array
+	softwareProductItems := make([]*models.SoftwareProductItem, 0, len(resp.Payload.SoftwareProducts))
+
+	for _, softwareProduct := range resp.Payload.SoftwareProducts {
+		softwareProductItems = append(softwareProductItems, &models.SoftwareProductItem{
+			APIGatewayUse:                  zero.BoolFrom(softwareProduct.APIGatewayUse),
```

Comment:
This conversion will lead to a `false` JSON value becoming `null` in the graphql response. It looks like the generated swagger code will always make a `false` from `null`, so I think the best thing here might be to just use `null.BoolFrom` instead.

Reply:
I made the change.

We also use zero.Bool in all the other CEDAR clients, I assume we want to change those as well?